File: src/main/java/Chapter02/iteration/FizzBuzz.java
package Chapter02.iteration;

public class FizzBuzz {

    public static void main(String[] args) {
        for (int i = 1; i <= 100; i++) {
            System.out.println(fizzBuzz(i));
        }
        for (int i = 100; i > 0; i -= 2) {
            System.out.println(fizzBuzz(i));
        }
    }

    private static String fizzBuzz(int i) {
        if (i % 15 == 0) {
            return "FizzBuzz";
        }
        if (i % 3 == 0) {
            return "Fizz";
        }
        if (i % 5 == 0) {
            return "Buzz";
        }
        return String.valueOf(i);
    }

}

File: src/main/kotlin/Chapter02/iteration/FizzBuzz.kt
package Chapter02.iteration

fun main() {
    for (i in 1..100) {
        println(fizzBuzz(i))
    }
    for (i in 100 downTo 1 step 2) {
        println(fizzBuzz(i))
    }
}

fun fizzBuzz(i: Int) =
    when {
        i % 15 == 0 -> "FizzBuzz"
        i % 3 == 0 -> "Fizz"
        i % 5 == 0 -> "Buzz"
        else -> "$i"
    }